samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -315,7 +319,7 @@ void link_ind_handler(const struct lte_lc_evt *const evt)
 		}
 
 		for (i = 0; i < cells.ncells_count; i++) {
-			/* Neighbor cells: */
+			/* Actual neighbor cells of a current cell: */
```

Comment:
On line 318, you could be more specific about what is not found ( e.g. "No legacy PCI neighbor cell information from modem"). I found it confusing to see this message but still get a list of (GCI) neighbours.

Reply:
Understand the point, but not sure about the division between PCI and GCI. I think that in network, both are actually having both GCI and PCI, eventhough GCI not returned by modem for the "actual neighbors". 

In practice the ncells_count is always 0 for other than the serving/current cell, i.e. no neighbor_cells list is available.
What about if printing `No neighbor cell information for the current cell from modem`? @tokangas or @trantanen, do you have any comments on this?

Reply:
I think we need to come up with terminology for the PCI and GCI neighbors (we know what they mean but most of the people don't know), and then write it down into lte_lc api and potentially its documentation as well as into mosh documentation.
Once that's done, it's easier to write error messages and name variables etc.

Reply:
All right, aligning terminology with the latest AT command spec. They are now talking about "surrounding cell" for these new guys.
So, I kept printing `No neighbor cell information from modem.` for current cell as it has been.

And for GCI search types there will be printing:
`Surrounding cell X from GCI search`

Full example:
`mosh:~$ link ncellmeas --single --search_type gci_2
Neighbor cell measurements and reporting starting
RRC mode: Idle
Neighbor cell measurement results:
  Current cell:
    ID 1805086, phy ID 110, MCC 244 MNC 12, RSRP 54 : -86dBm, RSRQ 24, TAC 312, earfcn 6200, meas time 22680
    TA 49, TA meas time 22616
  No neighbor cell information from modem.
  Surrounding cell 1 from GCI search
    ID 2200074, phy ID 465, MCC 244 MNC 12, RSRP 49 : -91dBm, RSRQ 14, TAC 322, earfcn 6200, meas time 22680
    TA "not valid", TA meas time 0
  Surrounding cell 2 from GCI search
    ID 1669123, phy ID 251, MCC 244 MNC 5, RSRP 41 : -99dBm, RSRQ 22, TAC 23040, earfcn 1825, meas time 22909
    TA "not valid", TA meas time 0
  Surrounding cell 3 from GCI search
    ID 37154335, phy ID 449, MCC 244 MNC 91, RSRP 53 : -87dBm, RSRQ 24, TAC 4020, earfcn 1300, meas time 22919
    TA "not valid", TA meas time 0
  Surrounding cell 4 from GCI search
    ID 37372427, phy ID 75, MCC 244 MNC 91, RSRP 42 : -98dBm, RSRQ 9, TAC 4020, earfcn 1300, meas time 22919
    TA "not valid", TA meas time 0
`

Reply:
Terminology is fine for me but I think we should document what `neighbor cell` and `surrounding cell` mean (and maybe mention also `current cell` so people have a list of all reported cells documented) in both LTE LC and MOSH documentation.
Given this stuff is not fully mature (although I don't know if the terminology will change), we could think of the documentation later as well.

Reply:
My favorite would have been the following, sorry for not following up in time:

"Measurements for cells in neighbor cell list" versus "Measurements for surrounding cells"